pkg/table-rule-selector/trie_selector.go

```diff
@@ -130,62 +229,116 @@ func (t *trieSelector) insertTable(schema, table string, rule interface{}, repla
 		return errors.Annotate(err, "insert into schema selector")
 	}
 
-	if schemaEntity.nextLevel == nil {
-		schemaEntity.nextLevel = newNode()
+	if schemaEntity.getNextLevel() == nil {
+		schemaEntity.setNextLevel(newNode())
 	}
 
-	_, err = t.insert(schemaEntity.nextLevel, table, rule, replace)
+	_, err = t.insert(schemaEntity.getNextLevel(), table, rule, replace)
 	if err != nil {
 		return errors.Annotate(err, "insert into table selector")
 	}
 
 	return nil
 }
 
+func (t *trieSelector) getRangeItem(pattern string) (*rangeItem, int) {
+	nextI := -1
+	for i := range pattern {
+		if pattern[i] == rangeClose {
+			nextI = i
+			break
+		}
+	}
+	if nextI == -1 {
+		return nil, nextI
+	}
+	item := &rangeItem{}
+	startI := 1
+	if pattern[startI] == rangeNot {
+		startI++
+		item.hasNot = true
+	}
+	for i := startI; i < nextI; i++ {
+		if pattern[i+1] == rangeBetween {
+			item.ranges = append(item.ranges, ran{pattern[i], pattern[i+2], true})
```

Comment:
`ran{pattern[i], pattern[i=2]` does not handle un-paired between such as `"ik[A-]"` very well, which will be recognized as `ik[A-]]`
Maybe we can make regulation about how the range selector works in some corner case, such as `[]` matches nothing, `[` matches `[`, `[!]` matches everything, `[a-]` is not determined

Reply:
Is `ik[A-]` a correct range? I just try it in "vim". It was recognized as `ik[A-A\--\-]`.

Reply:
And `[!]` is recognized as `[\!-\!]`.

Reply:
Oh, sorry, I misunderstood your mean. I'll change by your advice.